### PR TITLE
Run edit and execute, first version

### PR DESCRIPTION
**Pull request by OlehKSS**
The files I've been working on lately. They are not conflict with older, but in oder to make them useful we need to remove older or change files.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/109%23issuecomment-170839542%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/109%23issuecomment-170854465%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/109%23issuecomment-170864034%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/109%23discussion_r49673429%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/109%23discussion_r49673777%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/109%23discussion_r49674046%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/109%23discussion_r49674214%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/109%23discussion_r49674320%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/109%23discussion_r49674446%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/109%23discussion_r49674851%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/109%23issuecomment-172007286%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/109%23issuecomment-172107998%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/109%23issuecomment-172135619%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/109%23issuecomment-172307914%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/109%23issuecomment-172308901%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/109%23issuecomment-172312965%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/109%23issuecomment-172563667%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/109%23issuecomment-170839542%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22it%20seems%20that%20you%27ve%20forgotten%20to%20delete%20%60runs/execute%60%20route%20from%20%60runs%60%20module.%20I%20still%20see%20the%20old%20page%22%2C%20%22created_at%22%3A%20%222016-01-12T08%3A40%3A06Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22yes%2C%20I%20have.%20But%20there%20are%20several%20files%20to%20be%20changed%2C%20not%20only%20the%20rout.%5Cr%5Cn%22%2C%20%22created_at%22%3A%20%222016-01-12T09%3A46%3A22Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/16047980%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/OlehKSS%22%7D%7D%2C%20%7B%22body%22%3A%20%22Try%20it%20now.%20I%27ve%20connected%20the%20router.%22%2C%20%22created_at%22%3A%20%222016-01-12T10%3A16%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/16047980%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/OlehKSS%22%7D%7D%2C%20%7B%22body%22%3A%20%22Ok.%20%20I%20hope%20I%20will%20meet%20the%20above%20requirements.%22%2C%20%22created_at%22%3A%20%222016-01-15T16%3A28%3A42Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/16047980%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/OlehKSS%22%7D%7D%2C%20%7B%22body%22%3A%20%22Actually%20the%20task%20was%20to%20redesign%20test%20execution%20page.%20And%20it%20is%20done.%5Cr%5CnNow%20we%20shall%20move%20to%20the%20next%20task%20with%20routing%20enhancement.%5Cr%5Cn%5Cr%5CnPlease%20squash%20commits%20into%201%20and%20I%20will%20merge.%5Cr%5CnBe%20ready%20that%20logic%20will%20be%20changed%20almost%20completely%20here.%20So%20we%20have%20no%20need%20to%20review%20the%20code%20of%20controllers%20so%20deep.%20%22%2C%20%22created_at%22%3A%20%222016-01-15T22%3A08%3A38Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22What%20have%20I%20done%20wrong%3F%5Cr%5Cn%22%2C%20%22created_at%22%3A%20%222016-01-16T00%3A25%3A19Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/16047980%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/OlehKSS%22%7D%7D%2C%20%7B%22body%22%3A%20%22just%20merge%20again%20with%20master%20and%20read%20my%20skype%20message%20%3A%29%22%2C%20%22created_at%22%3A%20%222016-01-17T10%3A05%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40OlehKSS%20and%20after%20%60gulp%20test%60%20task%20will%20not%20fail%20you%20need%20to%20push%201%20commit%20instead%20of%2016%22%2C%20%22created_at%22%3A%20%222016-01-17T10%3A07%3A11Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22OK.%20let%27s%20try%5Cr%5Cn%22%2C%20%22created_at%22%3A%20%222016-01-17T10%3A48%3A54Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/16047980%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/OlehKSS%22%7D%7D%2C%20%7B%22body%22%3A%20%22%23%23%20%5BCurrent%20coverage%5D%5B1%5D%20is%20%6047.24%25%60%5Cn%3E%20Merging%20%2A%2A%23109%2A%2A%20into%20%2A%2Amaster%2A%2A%20will%20decrease%20coverage%20by%20%2A%2A-0.13%25%2A%2A%20as%20of%20%5B%60a77aff2%60%5D%5B3%5D%5Cn%5Cn%5Cn%60%60%60diff%5Cn%40%40%20%20%20%20%20%20%20%20%20%20%20%20master%20%20%20%20%23109%20%20%20diff%20%40%40%5Cn%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%5Cn%20%20Files%20%20%20%20%20%20%20%20%20%20%2061%20%20%20%20%20%2065%20%20%20%20%20%2B4%5Cn%20%20Stmts%20%20%20%20%20%20%20%20%201028%20%20%20%201069%20%20%20%20%2B41%5Cn%20%20Branches%20%20%20%20%20%20%20%20%200%20%20%20%20%20%20%200%20%20%20%20%20%20%20%5Cn%20%20Methods%20%20%20%20%20%20%20%20%20%200%20%20%20%20%20%20%200%20%20%20%20%20%20%20%5Cn%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%5Cn%2B%20Hit%20%20%20%20%20%20%20%20%20%20%20%20487%20%20%20%20%20505%20%20%20%20%2B18%5Cn%20%20Partial%20%20%20%20%20%20%20%20%20%200%20%20%20%20%20%20%200%20%20%20%20%20%20%20%5Cn-%20Missed%20%20%20%20%20%20%20%20%20541%20%20%20%20%20564%20%20%20%20%2B23%5Cn%60%60%60%5Cn%5Cn%3E%20Review%20entire%20%5BCoverage%20Diff%5D%5B4%5D%20as%20of%20%5B%60a77aff2%60%5D%5B3%5D%5Cn%5Cn%5Cn-----%5Cn%23%23%23%20%5BUncovered%20Suggestions%5D%5B2%5D%5Cn%5Cn1.%20%60%2B1.49%25%60%20via%20%5B...ake-tests.factory.js%23115...130%5D%28https%3A//codecov.io/github/ITsvetkoFF/Kv-012/src/client/app/blocks/fakers/fake-tests.factory.js%3Fref%3Da77aff25536c795d6d41af98d39c6519491ab761%23115%29%20%5Cn1.%20%60%2B1.49%25%60%20via%20%5B...create.controller.js%2370...85%5D%28https%3A//codecov.io/github/ITsvetkoFF/Kv-012/src/client/app/tests/tests-create.controller.js%3Fref%3Da77aff25536c795d6d41af98d39c6519491ab761%2370%29%20%5Cn1.%20%60%2B1.21%25%60%20via%20%5B...create.controller.js%2311...23%5D%28https%3A//codecov.io/github/ITsvetkoFF/Kv-012/src/client/app/tests/tests-create.controller.js%3Fref%3Da77aff25536c795d6d41af98d39c6519491ab761%2311%29%20%5Cn1.%20%2A%5BSee%207%20more...%5D%5B2%5D%2A%5Cn%5Cn%5B1%5D%3A%20https%3A//codecov.io/github/ITsvetkoFF/Kv-012%3Fref%3Da77aff25536c795d6d41af98d39c6519491ab761%5Cn%5B2%5D%3A%20https%3A//codecov.io/github/ITsvetkoFF/Kv-012/features/suggestions%3Fref%3Da77aff25536c795d6d41af98d39c6519491ab761%5Cn%5B3%5D%3A%20https%3A//codecov.io/github/ITsvetkoFF/Kv-012/commit/a77aff25536c795d6d41af98d39c6519491ab761%5Cn%5B4%5D%3A%20https%3A//codecov.io/github/ITsvetkoFF/Kv-012/compare/f91cb8b5463187bd74db38d19c4f097f384a3029...a77aff25536c795d6d41af98d39c6519491ab761%5Cn%5Cn%3E%20Powered%20by%20%5BCodecov%5D%28https%3A//codecov.io%29.%20Updated%20on%20successful%20CI%20builds.%22%2C%20%22created_at%22%3A%20%222016-01-18T15%3A37%3A21Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8655789%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/codecov-io%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20d35194dc90cdd84b3e38549f592fb9de8d0a3b16%20TCMSApp/src/client/app/runs/runs-exec/runs-execute.controller.js%2020%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/109%23discussion_r49674851%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22styles%20here%20too%3F%5Cr%5Cnhave%20you%20considered%20having%20the%20switch%20between%20run%20and%20edit%20outside%20the%20ui-view%20with%20run%20and%20edit%3F%5Cr%5Cnto%20be%20discussed%22%2C%20%22created_at%22%3A%20%222016-01-14T00%3A54%3A43Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20TCMSApp/src/client/app/runs/runs-exec/runs-execute.controller.js%3AL1-50%22%7D%2C%20%22Pull%20d35194dc90cdd84b3e38549f592fb9de8d0a3b16%20TCMSApp/src/client/app/runs/runs-exec/runs-exec.route.js%2027%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/109%23discussion_r49674446%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22where%20do%20you%20use%20these%20settings%3F%22%2C%20%22created_at%22%3A%20%222016-01-14T00%3A49%3A30Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20TCMSApp/src/client/app/runs/runs-exec/runs-exec.route.js%3AL1-62%22%7D%2C%20%22Pull%20d35194dc90cdd84b3e38549f592fb9de8d0a3b16%20TCMSApp/src/client/app/runs/runs-exec/runs-edit.html%2030%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/109%23discussion_r49674320%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%20don%27t%20understand%20that%20placeholders%20at%20all%21%22%2C%20%22created_at%22%3A%20%222016-01-14T00%3A47%3A41Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20TCMSApp/src/client/app/runs/runs-exec/runs-edit.html%3AL1-92%22%7D%2C%20%22Pull%20d35194dc90cdd84b3e38549f592fb9de8d0a3b16%20TCMSApp/src/client/app/runs/runs-exec/runs-edit.html%207%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/109%23discussion_r49674214%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22but%20we%20decided%20to%20use%20%60vmRunsEdit%60%20in%20HTML....%22%2C%20%22created_at%22%3A%20%222016-01-14T00%3A46%3A17Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20TCMSApp/src/client/app/runs/runs-exec/runs-edit.html%3AL1-92%22%7D%2C%20%22Pull%20d35194dc90cdd84b3e38549f592fb9de8d0a3b16%20TCMSApp/src/client/app/runs/runs-exec/runs-edit.controller.js%2022%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/109%23discussion_r49673429%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22you%27re%20setting%20styles%20in%20controller.%5Cr%5Cn%5Cr%5Cnmuch%20better%20solution%20is%20to%20set%20class%20here.%20and%20style%20the%20class%20in%20css.%22%2C%20%22created_at%22%3A%20%222016-01-14T00%3A35%3A03Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20TCMSApp/src/client/app/runs/runs-exec/runs-edit.controller.js%3AL1-50%22%7D%2C%20%22Pull%20d35194dc90cdd84b3e38549f592fb9de8d0a3b16%20TCMSApp/src/client/app/runs/runs-exec/runs-edit.controller.js%2017%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/109%23discussion_r49674046%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22will%20name%2C%20build%20and%20envShort%20be%20undefined%20without%20that%20lines%3F%20%3A%29%22%2C%20%22created_at%22%3A%20%222016-01-14T00%3A43%3A42Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20TCMSApp/src/client/app/runs/runs-exec/runs-edit.controller.js%3AL1-50%22%7D%2C%20%22Pull%20d35194dc90cdd84b3e38549f592fb9de8d0a3b16%20TCMSApp/src/client/app/runs/runs-exec/runs-edit.controller.js%2012%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/109%23discussion_r49673777%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22you%20shall%20write%20above%20this%20line%3A%5Cr%5Cn%5Cr%5Cn%60//TODO%3A%20change%20this%20temporary%20solution%20of%20exchanging%20run%20data%20between%20edit%20and%20run%20tabs%60%22%2C%20%22created_at%22%3A%20%222016-01-14T00%3A40%3A06Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20TCMSApp/src/client/app/runs/runs-exec/runs-edit.controller.js%3AL1-50%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-012/pull/109#issuecomment-170839542'>General Comment</a></b>
- <a href='https://github.com/ITsvetkoFF'></a> it seems that you've forgotten to delete `runs/execute` route from `runs` module. I still see the old page
- <a href='https://github.com/OlehKSS'></a> yes, I have. But there are several files to be changed, not only the rout.
- <a href='https://github.com/OlehKSS'></a> Try it now. I've connected the router.
- <a href='https://github.com/OlehKSS'></a> Ok.  I hope I will meet the above requirements.
- <a href='https://github.com/ITsvetkoFF'></a> Actually the task was to redesign test execution page. And it is done.
Now we shall move to the next task with routing enhancement.
Please squash commits into 1 and I will merge.
Be ready that logic will be changed almost completely here. So we have no need to review the code of controllers so deep.
- <a href='https://github.com/OlehKSS'></a> What have I done wrong?
- <a href='https://github.com/ITsvetkoFF'></a> just merge again with master and read my skype message :)
- <a href='https://github.com/ITsvetkoFF'></a> @OlehKSS and after `gulp test` task will not fail you need to push 1 commit instead of 16
- <a href='https://github.com/OlehKSS'></a> OK. let's try
- <a href='https://github.com/codecov-io'></a> ## [Current coverage][1] is `47.24%`
> Merging **#109** into **master** will decrease coverage by **-0.13%** as of [`a77aff2`][3]
```diff
@@            master    #109   diff @@
======================================
Files           61      65     +4
Stmts         1028    1069    +41
Branches         0       0
Methods          0       0
======================================
+ Hit            487     505    +18
Partial          0       0
- Missed         541     564    +23
```
> Review entire [Coverage Diff][4] as of [`a77aff2`][3]
-----
### [Uncovered Suggestions][2]
1. `+1.49%` via [...ake-tests.factory.js#115...130](https://codecov.io/github/ITsvetkoFF/Kv-012/src/client/app/blocks/fakers/fake-tests.factory.js?ref=a77aff25536c795d6d41af98d39c6519491ab761#115)
1. `+1.49%` via [...create.controller.js#70...85](https://codecov.io/github/ITsvetkoFF/Kv-012/src/client/app/tests/tests-create.controller.js?ref=a77aff25536c795d6d41af98d39c6519491ab761#70)
1. `+1.21%` via [...create.controller.js#11...23](https://codecov.io/github/ITsvetkoFF/Kv-012/src/client/app/tests/tests-create.controller.js?ref=a77aff25536c795d6d41af98d39c6519491ab761#11)
1. *[See 7 more...][2]*
[1]: https://codecov.io/github/ITsvetkoFF/Kv-012?ref=a77aff25536c795d6d41af98d39c6519491ab761
[2]: https://codecov.io/github/ITsvetkoFF/Kv-012/features/suggestions?ref=a77aff25536c795d6d41af98d39c6519491ab761
[3]: https://codecov.io/github/ITsvetkoFF/Kv-012/commit/a77aff25536c795d6d41af98d39c6519491ab761
[4]: https://codecov.io/github/ITsvetkoFF/Kv-012/compare/f91cb8b5463187bd74db38d19c4f097f384a3029...a77aff25536c795d6d41af98d39c6519491ab761
> Powered by [Codecov](https://codecov.io). Updated on successful CI builds.
- [ ] <a href='#crh-comment-Pull d35194dc90cdd84b3e38549f592fb9de8d0a3b16 TCMSApp/src/client/app/runs/runs-exec/runs-edit.controller.js 22'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-012/pull/109#discussion_r49673429'>File: TCMSApp/src/client/app/runs/runs-exec/runs-edit.controller.js:L1-50</a></b>
- <a href='https://github.com/ITsvetkoFF'></a> you're setting styles in controller.
much better solution is to set class here. and style the class in css.
- [ ] <a href='#crh-comment-Pull d35194dc90cdd84b3e38549f592fb9de8d0a3b16 TCMSApp/src/client/app/runs/runs-exec/runs-edit.controller.js 12'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-012/pull/109#discussion_r49673777'>File: TCMSApp/src/client/app/runs/runs-exec/runs-edit.controller.js:L1-50</a></b>
- <a href='https://github.com/ITsvetkoFF'></a> you shall write above this line:
`//TODO: change this temporary solution of exchanging run data between edit and run tabs`
- [ ] <a href='#crh-comment-Pull d35194dc90cdd84b3e38549f592fb9de8d0a3b16 TCMSApp/src/client/app/runs/runs-exec/runs-edit.controller.js 17'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-012/pull/109#discussion_r49674046'>File: TCMSApp/src/client/app/runs/runs-exec/runs-edit.controller.js:L1-50</a></b>
- <a href='https://github.com/ITsvetkoFF'></a> will name, build and envShort be undefined without that lines? :)
- [ ] <a href='#crh-comment-Pull d35194dc90cdd84b3e38549f592fb9de8d0a3b16 TCMSApp/src/client/app/runs/runs-exec/runs-edit.html 7'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-012/pull/109#discussion_r49674214'>File: TCMSApp/src/client/app/runs/runs-exec/runs-edit.html:L1-92</a></b>
- <a href='https://github.com/ITsvetkoFF'></a> but we decided to use `vmRunsEdit` in HTML....
- [ ] <a href='#crh-comment-Pull d35194dc90cdd84b3e38549f592fb9de8d0a3b16 TCMSApp/src/client/app/runs/runs-exec/runs-edit.html 30'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-012/pull/109#discussion_r49674320'>File: TCMSApp/src/client/app/runs/runs-exec/runs-edit.html:L1-92</a></b>
- <a href='https://github.com/ITsvetkoFF'></a> I don't understand that placeholders at all!
- [ ] <a href='#crh-comment-Pull d35194dc90cdd84b3e38549f592fb9de8d0a3b16 TCMSApp/src/client/app/runs/runs-exec/runs-exec.route.js 27'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-012/pull/109#discussion_r49674446'>File: TCMSApp/src/client/app/runs/runs-exec/runs-exec.route.js:L1-62</a></b>
- <a href='https://github.com/ITsvetkoFF'></a> where do you use these settings?
- [ ] <a href='#crh-comment-Pull d35194dc90cdd84b3e38549f592fb9de8d0a3b16 TCMSApp/src/client/app/runs/runs-exec/runs-execute.controller.js 20'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-012/pull/109#discussion_r49674851'>File: TCMSApp/src/client/app/runs/runs-exec/runs-execute.controller.js:L1-50</a></b>
- <a href='https://github.com/ITsvetkoFF'></a> styles here too?
have you considered having the switch between run and edit outside the ui-view with run and edit?
to be discussed

<a href='https://www.codereviewhub.com/ITsvetkoFF/Kv-012/pull/109?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/ITsvetkoFF/Kv-012/pull/109?approve=1'></a>&nbsp;<a href='https://github.com/ITsvetkoFF/Kv-012/pull/109'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by ITsvetkoFF** [comment]
it seems that you've forgotten to delete `runs/execute` route from `runs` module. I still see the old page

**Comment by OlehKSS** [comment]
yes, I have. But there are several files to be changed, not only the rout.

**Comment by OlehKSS** [comment]
Try it now. I've connected the router.

File: TCMSApp/src/client/app/runs/runs-exec/runs-edit.controller.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        vm.stateName = $state.current.name;
+        vm.addEnvDetail = addEnvDetail;
+        vm.cancel = cancel;
+        vm.save = save;
+        vm.name = undefined;//temporary run name
+        vm.build = undefined;//temporary run build
+        vm.envShort = undefined;//temporary run environment
+        vm.styles = {
+            backColorEdit: '#e6e6e6',
+            backColorRun: '#fff'
```
Comment (ITsvetkoFF) [inline TCMSApp/src/client/app/runs/runs-exec/runs-edit.controller.js]: you're setting styles in controller.

much better solution is to set class here. and style the class in css.

File: TCMSApp/src/client/app/runs/runs-exec/runs-edit.controller.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    angular
+        .module('app.runsExec')
+        .controller('RunsEditController', RunsEditController);
+
+    RunsEditController.$inject = ['$stateParams','$state', 'logger', 'moment'];
+
+    function RunsEditController($stateParams, $state) {
+        var vm = this;
+        vm.run = JSON.parse(JSON.stringify($stateParams.run));//creating of the deep copy
```
Comment (ITsvetkoFF) [inline TCMSApp/src/client/app/runs/runs-exec/runs-edit.controller.js]: you shall write above this line:

`//TODO: change this temporary solution of exchanging run data between edit and run tabs`

File: TCMSApp/src/client/app/runs/runs-exec/runs-edit.controller.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        vm.stateName = $state.current.name;
+        vm.addEnvDetail = addEnvDetail;
+        vm.cancel = cancel;
+        vm.save = save;
+        vm.name = undefined;//temporary run name
```
Comment (ITsvetkoFF) [inline TCMSApp/src/client/app/runs/runs-exec/runs-edit.controller.js]: will name, build and envShort be undefined without that lines? :)

File: TCMSApp/src/client/app/runs/runs-exec/runs-edit.html [hunk 4]
```diff
@@ -0,0 +1,91 @@
+<div class="matter">
+    <div class="container">
+        <div class="clearfix">
+            <h1>
+            <form class="form-inline pull-right">
+                <div class="btn-group">
+                    <button type="button" class="btn btn-default" ui-sref="runs-edit()" ng-style="{'background-color': vm.styles.backColorEdit}">Edit</button>
```
Comment (ITsvetkoFF) [inline TCMSApp/src/client/app/runs/runs-exec/runs-edit.html]: but we decided to use `vmRunsEdit` in HTML....

File: TCMSApp/src/client/app/runs/runs-exec/runs-edit.html [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+                <form action="" class="form">
+                    <div class="col-md-6">
+                    <div class="panel panel-info">
+                        <div class="panel-heading">
+                            Short information
+                            </div>
+                        <div class="panel-body">
+                    <div class="col-md-12">
+                    <label for="run-name">Name</label>
+                    <input type="text" name="" id="run-name" class="form-control" placeholder="{{vm.run.name||'Run name'}}" ng-model="vm.name"/>
```
Comment (ITsvetkoFF) [inline TCMSApp/src/client/app/runs/runs-exec/runs-edit.html]: I don't understand that placeholders at all!

File: TCMSApp/src/client/app/runs/runs-exec/runs-exec.route.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+                config: {
+                    url: '/runs/edit',
+                    templateUrl: 'app/runs/runs-exec/runs-edit.html',
+                    controller: 'RunsEditController',
+                    controllerAs: 'vm',
+                    title: 'Edit Test Run',
+                    settings: {
+                        nav: 5,
+                        content: '<i class="fa fa-team"></i> Edit Run'
+                    },
```
Comment (ITsvetkoFF) [inline TCMSApp/src/client/app/runs/runs-exec/runs-exec.route.js]: where do you use these settings?

File: TCMSApp/src/client/app/runs/runs-exec/runs-execute.controller.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+       var vm = this;
+       vm.run = JSON.parse(JSON.stringify($stateParams.run));//creating of the deep copy
+       vm.stateName = $state.current.name;
+       vm.progress = getProgress();
+       vm.suites = getSuites();
+       vm.selectedSuite = vm.suites[0];
+       vm.selectedTest = vm.run.tests[0];
+       vm.styles = {
+           backColorEdit: '#fff',
```
Comment (ITsvetkoFF) [inline TCMSApp/src/client/app/runs/runs-exec/runs-execute.controller.js]: styles here too?
have you considered having the switch between run and edit outside the ui-view with run and edit?
to be discussed

**Comment by OlehKSS** [comment]
Ok.  I hope I will meet the above requirements.

**Comment by ITsvetkoFF** [comment]
Actually the task was to redesign test execution page. And it is done.
Now we shall move to the next task with routing enhancement.

Please squash commits into 1 and I will merge.
Be ready that logic will be changed almost completely here. So we have no need to review the code of controllers so deep.

**Comment by OlehKSS** [comment]
What have I done wrong?

**Comment by ITsvetkoFF** [comment]
just merge again with master and read my skype message :)

**Comment by ITsvetkoFF** [comment]
@OlehKSS and after `gulp test` task will not fail you need to push 1 commit instead of 16

**Comment by OlehKSS** [comment]
OK. let's try

**Pull request merged**